Can't cross-compile to Windows with `module_mbedtls_enabled=no`

### Tested versions

43b32f9d0baa13ef8e7079521fbef7c4f090aa88 (latest)
* Introduced by #90482

### System information

Linux

### Issue description

When I cross-compile from Linux to Windows using `module_mbedtls_enabled=no`, I get the following errors:
```
In file included from thirdparty/mbedtls/include/mbedtls/build_info.h:174,
                 from thirdparty/mbedtls/library/common.h:14,
                 from thirdparty/mbedtls/library/aes.c:14:
thirdparty/mbedtls/include/mbedtls/check_config.h:28:2: error: #error "MBEDTLS_PLATFORM_C is required on Windows"
   28 | #error "MBEDTLS_PLATFORM_C is required on Windows"
      |  ^~~~~
In file included from thirdparty/mbedtls/include/mbedtls/build_info.h:174,
                 from thirdparty/mbedtls/library/common.h:14,
                 from thirdparty/mbedtls/library/aes.c:14:
thirdparty/mbedtls/include/mbedtls/check_config.h:578:2: error: #error "MBEDTLS_PLATFORM_SNPRINTF_ALT defined, but not all prerequisites"
  578 | #error "MBEDTLS_PLATFORM_SNPRINTF_ALT defined, but not all prerequisites"
      |  ^~~~~
thirdparty/mbedtls/include/mbedtls/check_config.h:592:2: error: #error "MBEDTLS_PLATFORM_VSNPRINTF_ALT defined, but not all prerequisites"
  592 | #error "MBEDTLS_PLATFORM_VSNPRINTF_ALT defined, but not all prerequisites"
      |  ^~~~~
```
Just compiling to Linux doesn't produce any errors.
This diff seems to fix it partially, though I get linker errors with undefined references:
```diff
diff --git a/thirdparty/mbedtls/include/godot_core_mbedtls_config.h b/thirdparty/mbedtls/include/godot_core_mbedtls_config.h
index d27bf608fbe..e65f332849f 100644
--- a/thirdparty/mbedtls/include/godot_core_mbedtls_config.h
+++ b/thirdparty/mbedtls/include/godot_core_mbedtls_config.h
@@ -33,6 +33,9 @@
 
 #include <limits.h>
 
+// Include default mbedTLS config.
+#include <mbedtls/mbedtls_config.h>
+
 // For AES
 #define MBEDTLS_CIPHER_MODE_CBC
 #define MBEDTLS_CIPHER_MODE_CFB
```
Copied from:
https://github.com/godotengine/godot/blob/578d937927ea9735aa7a8e0535f57801406b0b82/thirdparty/mbedtls/include/godot_module_mbedtls_config.h#L43-L44

Cross-compiling is probably unrelated to this, but I can't test it on Windows and thought it would be better to mention it.

### Steps to reproduce

Compile to Windows using `module_mbedtls_enabled=no`

--- a/thirdparty/mbedtls/include/godot_core_mbedtls_config.h
+++ b/thirdparty/mbedtls/include/godot_core_mbedtls_config.h
@@ -50,4 +50,11 @@
 #define MBEDTLS_PLATFORM_ZEROIZE_ALT
 #define MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES
 
+// This is only to pass a check in the mbedtls check_config.h header, none of
+// the files we include as part of the core build uses it anyway, we already
+// define MBEDTLS_PLATFORM_ZEROIZE_ALT which is the only relevant function.
+#if defined(__MINGW32__)
+#define MBEDTLS_PLATFORM_C
+#endif
+
 #endif // GODOT_CORE_MBEDTLS_CONFIG_H

